### DISCUSSION TITLE
[Bug] [Android] Entry MaterialEntryRenderer BoxBackgroundMode not working

### DISCUSSION THREAD

**Discussion by MhAllan** [xamarin/Xamarin.Forms#11082] [General]
### Description

I am trying to create custom renderer from the material renderer of Entry, trying to remove the terribly designed underline. So anything you could imagine or find online, from setting background to null to pain background drawable, .. etc .. all is not going to work. 

The only thing that has effect is `SetBoxBackgroundMode` but it is not working properly

### Steps to Reproduce
```csharp
public class CustomEntryRenderer : MaterialEntryRenderer
    {
        public CustomEntryRenderer(Context context) : base(context)
        {
        }
        protected override void OnElementChanged(ElementChangedEventArgs<Entry> e)
        {
            base.OnElementChanged(e);

            if(Control != null)
            {
                Control.SetBoxBackgroundMode(0);
            }
        }
    }
```

### Expected Behavior
To change entry border line

### Actual Behavior
setting it to zero, supposed to remove borders, it does nothing
setting it to 1, supposed to set the underline, it is there all the time without this setting
setting it to 2, supposed to make outlined, it outline the left, top, right sides and leaves the bottom not underlined

### Basic Information
XF 4.6.0.847
Android 9 and 10

**Comment by Mikilll94** [comment]
I also have this problem. I just want to have a **regular** entry from Material Design, NOT **filled**.

![Screenshot 2020-06-18 at 20 52 57](https://user-images.githubusercontent.com/17653648/85060656-e4871680-b1a5-11ea-9116-f247d26c7947.png)

**Comment by johnbmangold** [comment]
@Mikilll94 

if you use ```Control.SetBoxBackgroundMode(Google.Android.Material.TextField.TextInputLayout.BoxBackgroundNone);``` it'll give you the look you're going for.

**Comment by Mikilll94** [comment]
@johnbmangold 
Unfortunately, this is just removing the background color (which I can do also from XAML by setting the background color to white).

![Screenshot 2020-06-26 at 23 45 36](https://user-images.githubusercontent.com/17653648/85903647-2435aa00-b807-11ea-81c2-bf69ef54b36a.png)

However, the left, top, and bottom margins are still present. In Google Material Design there are many types of entries. I just want to use "regular" entry instead of "filled".

**Comment by averlius** [comment]
This is somehow related to something I'm looking for, I'm actually trying to have all the borders visible, but I just can't, even tried to create a shape and added to the control and nothing.

**Comment by talmurshidi** [comment]
If you are looking for something to remove the underline from entry then you can check this answer [remove entry underline](https://stackoverflow.com/a/68578038/13345950)
This is working with Xamarin Forms Visual Material 4.8+

```
[assembly: ExportRenderer(typeof(Entry), typeof(EntryMaterialRendererAndroid), new[] { typeof(VisualMarker.MaterialVisual) })]
namespace XFTest.Droid.Renderers
{
	public class EntryMaterialRendererAndroid : MaterialEntryRenderer
	{
		public EntryMaterialRendererAndroid(Context context) : base(context) { }
		protected override void OnElementChanged(ElementChangedEventArgs<Entry> e)
		{
			base.OnElementChanged(e);

			if (Control != null)
			{
				Control.BoxStrokeWidth = 0;
				Control.BoxStrokeWidthFocused = 0;
			}
		}
	}
}
```
